From: "Yan, Zheng" <zyan@redhat.com>
Date: Fri, 20 May 2016 15:41:20 +0800
Subject: ceph: avoid unnecessary fscache invalidation/revlidation
Git-commit: 1464975816c79a7cd28dc314384f060a122a9d55
Patch-mainline: v4.7-rc2
References: FATE#322288

ceph_fill_file_size() has already called ceph_fscache_invalidate()
if it return true.

Signed-off-by: Yan, Zheng <zyan@redhat.com>
Acked-by: Luis Henriques <lhenriques@suse.com>
---
 fs/ceph/caps.c | 9 +++------
 1 file changed, 3 insertions(+), 6 deletions(-)

diff --git a/fs/ceph/caps.c b/fs/ceph/caps.c
index c17b5d76d75e..7bdf7d59a36d 100644
--- a/fs/ceph/caps.c
+++ b/fs/ceph/caps.c
@@ -2993,10 +2993,9 @@ static void handle_cap_grant(struct ceph_mds_client *mdsc,
 	if (fill_inline)
 		ceph_fill_inline_data(inode, NULL, inline_data, inline_len);
 
-	if (queue_trunc) {
+	if (queue_trunc)
 		ceph_queue_vmtruncate(inode);
-		ceph_queue_revalidate(inode);
-	} else if (queue_revalidate)
+	else if (queue_revalidate)
 		ceph_queue_revalidate(inode);
 
 	if (writeback)
@@ -3199,10 +3198,8 @@ static void handle_cap_trunc(struct inode *inode,
 					  truncate_seq, truncate_size, size);
 	spin_unlock(&ci->i_ceph_lock);
 
-	if (queue_trunc) {
+	if (queue_trunc)
 		ceph_queue_vmtruncate(inode);
-		ceph_fscache_invalidate(inode);
-	}
 }
 
 /*

